torchdata/datapipes/iter/transform/callable.py

```diff
@@ -205,3 +206,162 @@ def __len__(self) -> int:
         if isinstance(self.datapipe, Sized):
             return len(self.datapipe)
         raise TypeError(f"{type(self).__name__} instance doesn't have valid length")
+
+
+@functional_datapipe("islice")
```

Comment:
`islice` might not be the best name. The behavior is different from [`itertolls.islice`](https://docs.python.org/3/library/itertools.html#itertools.islice). We might make users confused by the behavior.

Reply:
slice didn't seem quite right either, since it only slices the iterables within the datapipe, instead of the entire datapipe, but you're right the islice itertools version is a bit different too. Any other good names we might pick instead?

Reply:
@VitalyFedyunin just slice or should we call it something else?

Reply:
`slice` is good.